libbeat/autodiscover/appenders/feature.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package appenders
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/feature"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+// Namespace is the registry namespace for autodiscover appenders.
+var Namespace = "libbeat.autodiscover.appender"
+
+// Appender provides an interface by which extra configuration can be added into configs
+type Appender interface {
+	// Append takes a processed event and add extra configuration
+	Append(event bus.Event)
+}
+
+// Config settings
+type Config struct {
```

Comment:
We need to export this type?

Reply:
We have a type alias on the autodiscover package so we need to export it.